packages/utils/src/requestdata.ts

```diff
@@ -122,43 +124,52 @@ export function addRequestDataToTransaction(
 }
 
 /**
- * Extracts complete generalized path from the request object and uses it to construct transaction name.
+ * Extracts a complete and parameterized path from the request object and uses it to construct transaction name.
+ * If the parameterized transaction name cannot be extracted, we fall back to the raw URL.
+ *
+ * Additionally, this function determines and returns the transaction name source
  *
  * eg. GET /mountpoint/user/:id
  *
  * @param req A request object
  * @param options What to include in the transaction name (method, path, or both)
+ * @param customRoute An optional string that should be used as transaction name instead of the requests path
  *
- * @returns The fully constructed transaction name
+ * @returns A tuple of the fully constructed transaction name [0] and its source [1] (can be either route or url)
  */
 export function extractPathForTransaction(
   req: CrossPlatformRequest,
   options: { path?: boolean; method?: boolean } = {},
-): string {
+  customRoute?: string,
```

Comment:
IMHO it'd be better to add `customRoute` as a third option, rather than a third parameter. That way, if we were to, say, default the other two to `true` (which I also think we should do, though not in this PR), we wouldn't have to pass `undefined` as a placeholder in the argument list.

(Also, even leaving the other two options entirely aside, I have run into the double-optional-parameter problem a few times recently, and so I'm trying to be better about just always wrapping anything optional in an `options` object.)

Reply:
Good point, made the change to add the customRoute option to the options object.

> which I also think we should do, though not in this PR
agreed, that would be a good change